ingest/src/main/scala/geotrellis/osme/ingest/HttpSlippyTileReader.scala

```diff
@@ -0,0 +1,28 @@
+package geotrellis.osme.ingest
+
+import geotrellis.vector._
+import geotrellis.raster._
+import geotrellis.raster.io.geotiff._
+import geotrellis.spark._
+import geotrellis.spark.io.s3._
+import geotrellis.spark.io.slippy._
+import geotrellis.util.Filesystem
+
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.filefilter._
+import org.apache.commons.io.IOUtils._
+import org.apache.spark._
+import org.apache.spark.rdd._
+import java.net._
+import java.io.File
+
+class HttpSlippyTileReader[T](pathTemplate: String)(fromBytes: (SpatialKey, Array[Byte]) => T) extends SlippyTileReader[T] {
+    def getURL(template: String, z: Int, x: Int, y: Int) = 
+        template.replace("{z}", z.toString).replace("{x}", x.toString).replace("{y}", y.toString)
+    def getByteArray(url: String) = toByteArray(new URL(url).openStream())
```

Comment:
It's important to use try/finally when getting an input stream to close it after you're done with it. http://www.javapractices.com/topic/TopicAction.do?Id=25

https://github.com/geotrellis/geotrellis/blob/master/s3/src/main/scala/geotrellis/spark/io/s3/S3Client.scala#L166-L174